src/modules/iasql_functions/iasql.ts

```diff
@@ -1085,9 +1114,27 @@ async function commitApply(
   } while (ranFullUpdate);
   const t7 = Date.now();
   logger.info(`${dbId} applied and synced, total time: ${t7 - t1}ms`);
+  const out = iasqlPlanV3(toCreate, toUpdate, toReplace, toDelete);
+  if (recordsApplied > 0) {
+    const totalRecordsApplied = await MetadataRepo.incrementRecordsApplied(dbId, recordsApplied);
+    if (totalRecordsApplied) {
+      const user = await MetadataRepo.getUserFromDbId(dbId);
+      if (user) {
+        telemetry.logCommitApply(user?.id,
+        {
+          recordsApplied: totalRecordsApplied,
+          dbId,
+        },
+        {
+          output: out.rows.toString()
+        })
+      }
+    }
+
+  }
   await context.orm.dropConn();
   context.orm = oldOrm;
-  return iasqlPlanV3(toCreate, toUpdate, toReplace, toDelete);
+  return out;
```

Comment:
I think I'd rather this whole block live in an IIFE that's an async function so it doesn't block the main flow.